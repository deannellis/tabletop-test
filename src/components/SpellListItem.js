import React from 'react';

export default class SpellListItem extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            selected: ''
        }
    }
    
    render() {
        switch(this.props.spellType) {
            case 'magic-user':
                if(this.props.type == 'magic-user' || this.props.type == undefined){
                    return (
                        <div>
                            <label>
                                <input type="radio" name="class" value={this.props.id}
                                    checked={this.props.selected == this.props.id}
                                    onChange={this.handleOptionChange}
                                />
                                {this.props.name}
                            </label>
                            <p>{this.props.shortDescription}</p>
                        </div>
                    );
                } else {
                    return (<div></div>);
                }
                break;
            case 'cleric':
                if(this.props.type == 'cleric' || this.props.type == undefined){
                    return (
                        <div>
                            <label>
                                <input type="radio" name="class" value={this.props.id}
                                    checked={this.props.selected == this.props.id}
                                    onChange={this.handleOptionChange}
                                />
                                {this.props.name}
                            </label>
                            <p>{this.props.shortDescription}</p>
                        </div>
                    );
                } else {
                    return (<div></div>);
                }
                break;
            default:
                return (
                    <div>
                        <label>
                            <input type="radio" name="class" value={this.props.id}
                                    checked={this.props.selected == this.props.id}
                                    onChange={this.handleOptionChange}
                                />
                            {this.props.name}
                        </label>
                        <p>{this.props.shortDescription}</p>
                    </div>
                );
        }
    }
    
    handleOptionChange = (e) => {
        // console.log("this.state.selected:"+this.state.selected);
        // this.setState({
        //     selected: e.target.value
        // });
        // /\ This isn't doing anything. Make stateless
        this.props.handleSelectRadio(e.target.value);
    }
}