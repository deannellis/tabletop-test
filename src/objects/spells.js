export default [
    {
        id: '0000',
        level: 1,
        name: 'Charm Person',
        range: '30ft',
        type: 'magic-user',
        duration: 'special',
        shortDescription: 'This spell makes a humanoid creature regard the caster as its trusted friend and ally.',
        longDescription: 'This spell makes a humanoid creature of 4 hit dice or less regard the caster as its trusted friend and ally. And more...'
    },
    {
        id: '0001',
        level: 1,
        name: 'Detect Magic',
        range: '60ft',
        type: undefined,
        duration: '2 turns',
        shortDescription: 'The caster of this spell is able to detect magic objects or creatures.',
        longDescription: 'The caster of this spell is able to detect enchanted or enspelled objects or creatures within the given range by sight, seeing them surrounded by a pale glowing light. And more...'
    },
    {
        id: '0002',
        level: 1,
        name: 'Floating Disc',
        range: '10ft',
        type: 'magic-user',
        duration: '5 turns + 1/level',
        shortDescription: 'This spell creates an invisible plane of force for carrying loads.',
        longDescription: 'This spell creates an invisible, slightly concave circular plane of force for carrying loads. And more...'
    },
    {
        id: '0003',
        level: 1,
        name: 'Hold Portal',
        range: '100ft + 10ft/level',
        type: 'magic-user',
        duration: '1 round/level',
        shortDescription: 'This spell magically holds shut a portal (door, gate, window, etc)',
        longDescription: 'This spell magically holds shut a door, gate, window, or shutter of wood, metal, or stone. And more...'
    },
    {
        id: '0004',
        level: 1,
        name: 'Light',
        range: '120ft',
        type: undefined,
        duration: '6 turns + 1/level',
        shortDescription: 'This spell creates a light equal to torchlight',
        longDescription: 'This spell creates a light equal to torchlight which illuminates a 30ft radius area (and provides dim light for an additional 20) around the target location or object. And more...'
    },
    {
        id: '0005',
        level: 1,
        name: 'Cure Light Wounds',
        range: 'touch',
        type: 'cleric',
        duration: 'instantaneous',
        shortDescription: 'The caster heals 1d6+1 hit points of damage',
        longDescription: 'With this spell the caster heals 1d6+1 hit points of damage by laying his or her hand upon the injured creature.'
    }
]

// const sampleSpell = {
//     id: 0000,
//     level: 1,
//     name: 'Floating Disc',
//     range: '0',
//     type: 'magic-user',
//     duration: '5 turns + 1/level',
//     shortDescription: 'This spell creates an invisible, slightly concave circular plane of force for carrying loads.',
//     longDescription: 'This spell creates an invisible, slightly concave circular plane of force for carrying loads.'
// }