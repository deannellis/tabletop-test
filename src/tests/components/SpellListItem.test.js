/* global expect */
import React from 'react';
import { shallow } from 'enzyme';
import SpellListItem from '../../components/SpellListItem';
import spells from '../../objects/spells';

test('should render spell list item with cleric prop', () => {
    const wrapper = shallow(<SpellListItem spellType={'cleric'} {...spells[5] } />);
    expect(wrapper).toMatchSnapshot();
});

test('should render spell list item with magic-user prop', () => {
    const wrapper = shallow(<SpellListItem spellType={'magic-user'} {...spells[0] } />);
    expect(wrapper).toMatchSnapshot();
});

test('should render spell list item regardless of spellType prop', () => {
    const wrapper = shallow(<SpellListItem spellType={null} {...spells[1] } />);
    expect(wrapper).toMatchSnapshot();
});
