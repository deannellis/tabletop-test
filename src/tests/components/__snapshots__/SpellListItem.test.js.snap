// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render spell list item regardless of spellType prop 1`] = `
<div>
  <label>
    <input
      checked={false}
      name="class"
      onChange={[Function]}
      type="radio"
      value={1}
    />
    Detect Magic
  </label>
  <p>
    The caster of this spell is able to detect magic objects or creatures.
  </p>
</div>
`;

exports[`should render spell list item with cleric prop 1`] = `
<div>
  <label>
    <input
      checked={false}
      name="class"
      onChange={[Function]}
      type="radio"
      value={5}
    />
    Cure Light Wounds
  </label>
  <p>
    The caster heals 1d6+1 hit points of damage
  </p>
</div>
`;

exports[`should render spell list item with magic-user prop 1`] = `
<div>
  <label>
    <input
      checked={false}
      name="class"
      onChange={[Function]}
      type="radio"
      value={0}
    />
    Charm Person
  </label>
  <p>
    This spell makes a humanoid creature regard the caster as its trusted friend and ally.
  </p>
</div>
`;
